src/components/MapSelect/index.js

```diff
@@ -127,54 +147,55 @@ class MapSelect extends React.Component {
       position: 'topleft',
       element: div,
     });
-    loadingControl.addTo(this.map);
-
-    errorControl.addTo(this.map);
-  }
-
-  componentDidUpdate(prevProps) {
-    const lat = this.props.latlng.latitude;
-    const lng = this.props.latlng.longitude;
-    if (lat !== prevProps.latlng.latitude || lng !== prevProps.latlng.longitude) {
-      this.map.panTo({
-        lat,
-        lng,
-      });
-    }
 
-    const value = this.props.value;
-    if (isEqual(value, prevProps.value) === false) {
-      // Selection changed, update internal selection
-      this.selection.set.clear();
-      for (const id of value) {
-        this.selection.add(id);
-      }
-
-      // Let icons reflect new selection
-      // istanbul ignore next
-      for (const layer of this.featuresLayer.getLayers()) {
-        const properties = layer.feature.properties;
-        const id = properties[this.props.idField];
-        const iconType = properties[this.props.iconField];
-        const icon = this.props.getIcon(iconType, this.selection.has(id));
-        layer.setIcon(icon);
-      }
+    loadingControl.addTo(mapInstance);
+
+    errorControl.addTo(mapInstance);
+
+    return () => {
+      mapInstance.remove();
+    };
+    // only execute when the mapInstance is available; disabling linter
+    // eslint-disable-next-line
+  }, [mapInstance]);
+
+  useEffect(/* istanbul ignore next */ () => {
+    if (!featuresLayer.current) return;
+
+    selection.set.clear();
+
+    for (const id of value) {
+      selection.add(id);
     }
-  }
-
-  render() {
-    return (
-      <div className={classNames('map-component', { write: this.props.onSelectionChange })}>
-        <div className="map">
-          <div className="map-container" id="mapSelect" />
-        </div>
-      </div>
-    );
-  }
-}
+
+    // Let icons reflect new selection
+    featuresLayer.current.getLayers().forEach(layer => {
+      const properties = layer.feature.properties;
+      const id = properties[idField];
+      const iconType = properties[iconField];
+      const icon = getIcon(iconType, selection.has(id));
+
+      layer.setIcon(icon);
+    });
+    // only execute when value changes; disabling linter
+    // eslint-disable-next-line
+  }, [value]);
+
+  return (
+    <StyledMap
+      className={classNames('map-component', { write: onSelectionChange })}
+      data-testid="mapSelect"
+      mapOptions={mapOptions}
+      setInstance={setMapInstance}
+    >
+      <ViewerContainer data-testid="mapSelectViewerContainer">
+        <WfsLayer url={geojsonUrl} zoomLevel={16} />
```

Comment:
I don't think this is necessary, the implementation of the wfs functionality is done in BboxGeojsonLayer